### ISSUE TITLE
[eclipse ide] generated sources of model projects not correctly handled in eclipse

### ISSUE THREAD

**Issue by Hilbrand** [openhab/openhab-core#2121]
I tried out a new Eclipse with core. I used Eclipse 2020-12.

All the model projects (both `*.model` and `*.model.ide` have additional source directories: `xtend-gen` and `src-gen`. Both are not added as source projects or if added are set to `Excluded: **` (Which should be `None`). It looks like Eclipse is not picking up the `add-source` correctly. It might be a regression of the specific Eclipse version (as I've seen this behavior in older versions).

Manually fixing the missing directories in Eclipse works, until you do a maven refresh on a project. In that case it restores the invalid paths. One work-around is to simply remove the model projects from Eclipse, but that seems sub-optimal.

**Comment by wborn** [comment]
The openHAB Eclipse IDE is just doing what is instructed to do, which is [ignoring the build-helper-maven-plugin](https://github.com/openhab/openhab-distro/blob/1487aa17d91e879da902a2b74bbece97fdad35eb/launch/openHAB2.setup#L2395-L2407). See https://github.com/openhab/openhab-addons/issues/8641. I think that issue shouldn't have been closed by @kaikreuzer as the Oomph config hasn't been updated to unignore that plugin execution and [install the m2e connector](https://github.com/openhab/openhab-addons/issues/8641#issuecomment-703313315).

**Comment by wborn** [comment]
Can you check if this issue is fixed @Hilbrand now additional source dirs should be added in Eclipse by the builderhelper m2e plugin? (https://github.com/openhab/openhab-distro/pull/1259)

**Comment by bigbasec** [comment]
> Can you check if this issue is fixed @Hilbrand now additional source dirs should be added in Eclipse by the builderhelper m2e plugin? ([openhab/openhab-distro#1259](https://github.com/openhab/openhab-distro/pull/1259))

Yes, as long as you have the plugin enable it will build from a clean checkout.
